homeassistant/components/proxmoxve/__init__.py

```diff
@@ -86,93 +86,103 @@
 
 async def async_setup(hass: HomeAssistant, config: dict):
     """Set up the platform."""
-    hass.data.setdefault(DOMAIN, {})
+
+    hass.data[DOMAIN] = {}
+
+    # import to config flow
+    if DOMAIN in config:
+        for conf in config[DOMAIN]:
+            hass.async_create_task(
+                hass.config_entries.flow.async_init(
+                    DOMAIN, context={"source": SOURCE_IMPORT}, data=conf
+                )
+            )
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, config_entry):
+    """Set up the platform."""
+
+    entry_data = config_entry.data
 
     def build_client() -> ProxmoxAPI:
```

Comment:
This in-line method can be removed, and unindent all of its contents to make it just part of the entry setup. Reasoning: The setup entry is called for each entry already (which is why you have removed the iteration)